From 5c00c9c562803446b39e254fb477cd4fdbae374f Mon Sep 17 00:00:00 2001
From: Gordon Sim <gsim@redhat.com>
Date: Mon, 31 Mar 2014 17:35:04 +0100
Subject: [PATCH] Ensure routing key is specified in the address for a direct
 producer

This then matches the address used a direct consumer.

Change-Id: If71f78e50f8a9b3acfd1e9d02c8271f17c4ebee7
Closes-Bug: #1300318
(cherry picked from commit 9e0e85ede8406ace45c7573b18b8fe86c3e58921)
---
 oslo/messaging/_drivers/impl_qpid.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/oslo/messaging/_drivers/impl_qpid.py b/oslo/messaging/_drivers/impl_qpid.py
index 54efc5d..41e780e 100644
--- a/oslo/messaging/_drivers/impl_qpid.py
+++ b/oslo/messaging/_drivers/impl_qpid.py
@@ -378,7 +378,7 @@ class DirectPublisher(Publisher):
         """Init a 'direct' publisher."""
 
         if conf.qpid_topology_version == 1:
-            node_name = msg_id
+            node_name = "%s/%s" % (msg_id, msg_id)
             node_opts = {"type": "direct"}
         elif conf.qpid_topology_version == 2:
             node_name = "amq.direct/%s" % msg_id
-- 
1.7.7.6

